spynnaker/pyNN/models/neuron/structural_plasticity/synaptogenesis/elimination/random_by_weight_elimination.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from .abstract_elimination import AbstractElimination
+from pacman.model.decorators.overrides import overrides
+
+
+class RandomByWeightElimination(AbstractElimination):
+    """ Elimination Rule that depends on the weight of a synapse
+    """
+
+    __slots__ = [
+        "__prob_elim_depressed",
+        "__prob_elim_potentiatiated",
+        "__threshold"
+    ]
+
+    def __init__(
+            self, threshold, prob_elim_depressed=0.0245,
+            prob_elim_potentiatiated=1.36 * 10 ** -4):
+        """
+
+        :param threshold:\
+            Below this weight is considered depression, above or equal to this\
```

Comment:
weight is considered depressed  and .... is considered potentiated

Reply:
Not sure I see what the issue is here?